src/components/step-flow/step-flow.component.tsx

```diff
@@ -0,0 +1,193 @@
+import React from "react";
+import { MarginProps } from "styled-system";
+import Icon from "../icon";
+import IconButton from "../icon-button";
+import {
+  StyledStepFlow,
+  StyledStepContent,
+  StyledStepContentText,
+  StyledCategoryText,
+  StyledTitleText,
+  StyledTitleTextContainer,
+  StyledVisibleTitleText,
+  StyledHiddenTitleText,
+  StyledStepLabelAndProgress,
+  StyledStepLabel,
+  StyledProgressIndicatorBar,
+  StyledProgressIndicator,
+} from "./step-flow.style";
+import tagComponent, {
+  TagProps,
+} from "../../__internal__/utils/helpers/tags/tags";
+import { VariantTypes } from "../typography";
+import useLocale from "../../hooks/__internal__/useLocale";
+
+export type Steps = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8;
+
+export interface StepFlowProps extends MarginProps, TagProps {
+  /** A category for the user journey.  */
+  category?: string;
+  /** Set the variant of the internal 'Typography' component which contains the category.
+   * However, despite the chosen variant the styling will always be overridden.
+   */
+  categoryTypographyVariant?: VariantTypes;
+  /** The title of the current step.  */
+  title: string;
+  /** Set the variant of the internal 'Typography' component which contains the title.
+   * However, despite the chosen variant the styling will always be overridden.
+   */
+  titleTypographyVariant?: VariantTypes;
+  /** A ref which will be applied to the DOM element which contains the component's title.
+   * Used to focus on the title, to ensure screen reader user's are aware of title changes,
+   * and allow navigation from the StepFlow component to the contents below.
+   */
+  titleRef: React.RefObject<HTMLDivElement>;
+  /** The tabindex which will be applied to the DOM element which contains the component's title.
+   * This ensures the element becomes tabbable.
+   */
+  titleTabIndex?: -1 | 0;
+  /** The total steps in the user journey.  */
+  totalSteps: Steps;
+  /**
+   * The current step of the user journey. If the set `currentStep` is higher than
+   * `totalSteps`the value of `currentStep` will be that of `totalSteps` instead.
+   */
+  currentStep: Steps;
+  /** Determines if the progress indicator is shown. */
+  showProgressIndicator?: boolean;
+  /** Determines if the close icon button is shown */
+  showCloseIcon?: boolean;
+  /** function runs when user click dismiss button */
+  onDismiss?: (
+    e:
+      | React.KeyboardEvent<HTMLButtonElement>
+      | React.MouseEvent<HTMLButtonElement>
+  ) => void;
+}
+
+let currentStepWarnTriggered = false;
```

Comment:
comment: should this be within the component as in theory we'd want it to apply for any incorrect implementation of `StepFlow` not just once for all of them. We don't want it to repeatedly warn for the same one so the flag is needed